engines/dradis-api/app/controllers/dradis/ce/api/v1/comments_controller.rb

```diff
@@ -11,27 +11,46 @@ def index
       def show; end
 
       def create
+        @comment = @commentable.comments.build(comment_params)
+        @comment.user = current_user
+        if @comment.save
+          track_created(@comment)
+          render status: 201, location: @commentable
+        else
+          render_validation_errors(@comment)
+        end
       end
 
       def update
+        if @comment.update_attributes(comment_params)
+          track_updated(@comment)
+        else
+          render_validation_errors(@comment)
+        end
       end
 
       def destroy
+        @comment.destroy
+        track_destroyed(@comment)
+        render_successful_destroy_message
       end
 
       private
 
+      def comment_params
+        params.require(:comment).permit(:content)
+      end
+
       def set_commentable
-        @commentable =
-          if params[:issue_id]
-           Issue.find(params[:issue_id])
-          elsif params[:note_id]
-            Note.find(params[:note_id])
-          elsif params[:evidence_id]
-            Evidence.find(params[:evidence_id])
-          else
-            raise 'Polymorphic model missing!'
-          end
+        begin
+          id_param = params.keys.select { |key| key =~ /_id$/ }.first
+          id_value = params[id_param]
+
+          klass = id_param.chomp('_id').capitalize.constantize
```

Comment:
I'm a little uncertain with having the params being constantized. Is this exploitable? For example: send out a params with user_id to manipulate the User model?

Reply:
You have a point.
Reverting.